common/quotas/dynamicratelimiter.go

```diff
@@ -24,48 +24,103 @@ package quotas
 
 import (
 	"context"
+	"math"
+	"sync/atomic"
+	"time"
 
 	"golang.org/x/time/rate"
 
 	"github.com/uber/cadence/common/clock"
 )
 
+const _ttl = time.Second * 5
+const _minBurst = 1
+
 // DynamicRateLimiter implements a dynamic config wrapper around the rate limiter,
 // checks for updates to the dynamic config and updates the rate limiter accordingly
 type DynamicRateLimiter struct {
-	rps RPSFunc
-	rl  *RateLimiter
+	rps            RPSFunc
+	rl             clock.Ratelimiter
+	timeSource     clock.TimeSource
+	ttl            time.Duration
+	lastUpdateTime atomic.Value
+	minBurst       int
+}
+
+type DynamicRateLimiterOpt = func(limiter *DynamicRateLimiter)
+
+func WithTTL(ttl time.Duration) DynamicRateLimiterOpt {
+	return func(limiter *DynamicRateLimiter) {
+		limiter.ttl = ttl
+	}
```

Comment:
tbh I think func-closure-options are the worst option :\  they can't be introspected for tests or logs or comparison, and they're extremely hard to discover or go-to-source/impl/etc due to the weak typing they force.

could we change this to just an opts-struct?  static typing is a heck of a lot more useful for stuff like this, we don't need years-long-API-stability-at-the-cost-of-ergonomics stuff.

Reply:
Done